@timja can you test the commit I just pushed: https://github.com/timja/jenkins-ci.org-docker/pull/1/commits/5bd768dea05555e98493c2526c7365ba78b719c3 

I got the following results when trying an image with the "wrong" platform specified:

```shell
make build-alpine_jdk8 ARCH=arm64
+ docker buildx bake -f docker-bake.hcl --load linux --print
+ jq -r '.target | path(.. | select(.platforms[] | contains("linux/arm64"))?) | add'
Error: the image 'alpine_jdk8' does not exist in manifest for the platform 'linux/arm64'. Please check the output of 'make list'. Exiting.
make: *** [build-alpine_jdk8] Error 1
```

Of course, it builds successfully when the ARCH is ok:

```shell
make build-alpine_jdk8 ARCH=amd64  
+ docker buildx bake -f docker-bake.hcl --load linux --print
+ jq -r '.target | path(.. | select(.platforms[] | contains("linux/amd64"))?) | add'
+ docker buildx bake -f docker-bake.hcl --load --set '*.platform=linux/amd64' alpine_jdk8
#2 [internal] load .dockerignore
#2 sha256:38e2dddc2095909f88a98a84a202bef7e84f8d00471945908bf727dab8e65d5d
# ...
```

The tests are broken though: a 2nd commit is needed to update the way `bats` retrieve the image name

--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@ export DOCKER_CLI_EXPERIMENTAL=enabled
 export BUILDKIT_PROGRESS=plain
 
 current_arch := $(shell uname -m)
-export ARCH ?= $(shell case $(current_arch) in (x86_64) echo "amd64" ;; (i386) echo "386";; (aarch64) echo "arm64" ;; (armv6*) echo "arm/v6";; (armv7*) echo "arm/v7";; (ppc64*|s390*|riscv*) echo $(current_arch);; (*) echo "UNKNOWN-CPU";; esac)
+export ARCH ?= $(shell case $(current_arch) in (x86_64) echo "amd64" ;; (i386) echo "386";; (aarch64|arm64) echo "arm64" ;; (armv6*) echo "arm/v6";; (armv7*) echo "arm/v7";; (ppc64*|s390*|riscv*) echo $(current_arch);; (*) echo "UNKNOWN-CPU";; esac)
 
 all: shellcheck build test
 
@@ -47,8 +47,11 @@ build-%: check-reqs
 	@$(call check_image,$*)
 	@set -x; $(bake_base_cli) --set '*.platform=linux/$(ARCH)' '$*'
 
+show:
+	@$(bake_base_cli) linux --print
+
 list: check-reqs
-	@set -x; $(bake_base_cli) linux --print | jq -r '.target | path(.. | select(.platforms[] | contains("linux/$(ARCH)"))?) | add'
+	@set -x; make --silent show | jq -r '.target | path(.. | select(.platforms[] | contains("linux/$(ARCH)"))?) | add'
 
 bats:
 	git clone https://github.com/bats-core/bats-core bats ;\
@@ -159,4 +162,4 @@ clean:
 	rm -rf tests/test_helper/bats-*; \
 	rm -rf bats
 
-.PHONY: shellcheck check-reqs build clean test list test-install-plugins
+.PHONY: shellcheck check-reqs build clean test list test-install-plugins show

--- a/tests/test_helpers.bash
+++ b/tests/test_helpers.bash
@@ -40,7 +40,7 @@ function get_sut_image {
     # Option --print for 'docker buildx bake' prints the JSON configuration on the stdout
     # Option --silent for 'make' suppresses the echoing of command so the output is valid JSON
     # The image name is the 1st of the "tags" array, on the first "image" found
-    make --silent build-"${IMAGE}" BAKE_ADDITIONAL_FLAGS=--print | jq -r 'first(.target[]).tags[0]'
+    make --silent show | jq -r ".target.${IMAGE}.tags[0]"
 }
 
 function get_test_image {

